docs/providers/provider-protocol-reference.md

```diff
@@ -0,0 +1,243 @@
+pub async fn make_http_request(request: HttpRequest) -> Result<HttpResponse, HttpRequestError>;
+```
+
+#### Description
+
+This is the most useful imported function, it allows the providers to make HTTP queries to
+remote hosts through the hosting runtime. A basic example of its usage can be seen in the
+tutorial to create a provider:
+
+```rust
+    let base_url: Url = "https://jsonplaceholder.typicode.com".parse().unwrap();
+    let url = base_url
+        .join("/users")
+        .unwrap();
+
+    let request = HttpRequest {
+        url,
+        headers: None,
+        method: HttpRequestMethod::Get,
+        body: None,
+    };
+
+    let response = make_http_request(request).await?;
+
+    serde_json::from_slice(&response.body).map_err(|e| Error::Deserialization { message: format!("Could not deserialize payload: {e:?}") })
+```
+
+### `log`
+
+#### Signature
+
+```rust
+pub fn log(message: String);
+```
+
+#### Description
+
+This function asks the hosting runtime to log a message. This will make your log
+messages appear in Studio or in the standard output of the Fiberplane Daemon
+instance. Its usage is often as simple as:
+
+```rust
+    let location = GeoLocation { latitude: 50.1234, longitude: 4.12084837 };
+    log(format!("Catnip: looking for the closest dispenser to {location:?} now..."));
+```
+
+### `random`
+
+#### Signature
+
+```rust
+pub fn random(len: u32) -> Vec<u8>;
+```
+
+#### Description
+
+Query the hosting runtime for a source of randomness. Note that there is no
+guarantee that the random values you obtain this way to be cryptographically
+secure: it is entirely dependent on the runtime implementation. You can see the
+implementation of randomness for the Fiberplane Daemon in the [Provider Runtime
+crate](https://github.com/fiberplane/fiberplane-rs/blob/6679fbd0f1cbdac7c57422ae699e12bb35bed71b/fiberplane-provider-protocol/fiberplane-provider-runtime/src/spec/mod.rs#L116)
+source code.
+
+For example, you could create a
+[getrandom](https://docs.rs/getrandom/latest/getrandom/macro.register_custom_getrandom.html#writing-a-custom-getrandom-implementation) provider using it
+
+```rust
+// In a helper crate like getrandom_fp_provider
+use getrandom::Error;
+
+pub fn use_fp_bindings(buf: &mut [u8]) -> Result<(), Error> {
+    let vals = random(buf.len());
+    buf.copy_from_slice(&vals);
+    Ok(())
+}
+
+// In your provider crate
+use getrandom_fp_provider::use_fp_bindings;
+use getrandom::register_custom_getrandom;
+
+register_custom_getrandom!(use_fp_bindings);
+```
+
+> Note: this example snippet might prove generic and useful enough to be upstreamed
+  by Fiberplane and available as a public helper crate. Stay tuned!
+
+### `now`
+
+#### Signature
+
+```rust
+pub fn now() -> Timestamp;
+```
+
+#### Description
+
+Returns a timestamp of the current time of execution. The usefulness of this
+timestamp comes from what you want to do with timers, or signature
+creation/verification schemes.
+
+## Exported functions
```

Comment:
I would list the exported functions before the imported functions, since I think they are the starting point when you first implement a provider.